frontend/src/components/common/authorized-menu/components/become-a-mentor/become-a-mentor.tsx

```diff
@@ -0,0 +1,67 @@
+import mentorImage from 'assets/img/mentor.png';
+import { FC } from 'common/types/types';
+import { Button, Image } from 'components/common/common';
+import { useAppDispatch, useAppSelector, useParams } from 'hooks/hooks';
+import { courseActions } from 'store/actions';
+
+import styles from './styles.module.scss';
+
+const BecomeAMentor: FC = () => {
+  const { isMentorButtonVisible, passedInterviewsCategoryIds, course, user } =
+    useAppSelector((state) => ({ ...state.course, ...state.auth }));
+  const dispatch = useAppDispatch();
+
+  const { id } = useParams();
+
+  const handleClick = (): void => {
+    if (!course || !user) {
+      return;
+    }
+
+    const isInterviewPassed = passedInterviewsCategoryIds.includes(
+      course.courseCategoryId,
+    );
+
+    if (isInterviewPassed) {
+      dispatch(
+        courseActions.createMentor({ courseId: Number(id), userId: user.id }),
+      );
+
+      return;
+    }
+
+    dispatch(
+      courseActions.createInterview({
+        categoryId: course.courseCategoryId,
+        intervieweeUserId: user.id,
+      }),
+    );
+  };
+
+  if (!isMentorButtonVisible) {
+    return null;
+  }
```

Comment:
```suggestion
```
add this condition where you render it. the same as  the `onClick` function

Reply:
@what1s1ove didn't quite understand about `onClick` part. Should I move the `onClick` function to the component, where I render `BecomeAMentor` and pass it as props?